src/core/indexedlist.h

```diff
@@ -13,30 +13,91 @@ class IndexedList
     explicit IndexedList() = default;
```

Comment:
IndexedList is really used as a map<int, ToxCall>, that's the easiest way to see it.
Each ToxCall has a unique int ID that toxcore gives it, we use that as the int key of the map. But since that ID is in the ToxCall structure, we don't really need to make a map that stores the keys, we can just make a vector and ask the ToxCall what its key (it's ID) is.
Since there are rarely more than a couple ToxCalls going on at the same time, the vector never has more than a couple elements, and just iterating over it is super fast (cache locality, predictors friendliness and all that). It also uses less memory than a map of hash table would, since it only has to store an array of the values instead of a a more complicated tree or hash table.

So I though it was simple enough to live, but after having to explain it I can see how it's confusing, we can absolutely remove it if it makes the code easier to understand. A map<int, ToxCall> will work just the same :)